kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -158,7 +158,14 @@ default String getSingular() {
   }
 
   static String getFullResourceName(Class<?> clazz) {
-    return getFullResourceName(getPlural(clazz), getGroup(clazz));
+    final String plural = getPlural(clazz);
+    final String group = getGroup(clazz);
+    if (plural == null || group == null) {
```

Comment:
Looking at getPlural, it seems like it would always return non-null value

Reply:
Indeed. Fixed, thanks!